pkg/manager/member/tidb_member_manager.go

```diff
@@ -208,6 +208,10 @@ func (tmm *tidbMemberManager) syncTiDBStatefulSetForTidbCluster(tc *v1alpha1.Tid
 	}
 
 	if tmm.autoFailover {
+		if tc.Spec.TiDB.Replicas == int32(0) && tc.Status.TiDB.FailureMembers != nil {
+			tmm.tidbFailover.Recover(tc)
+			*newTiDBSet.Spec.Replicas = int32(0)
```

Comment:
I think it is not necessary to mutate the replicas here, in the next round of reconciling, the desired replicas will calculated as 0 by nature.

Reply:
done